### DISCUSSION TITLE
Are `preinit/postinit` supported anymore?

### DISCUSSION THREAD

**Discussion by TWolversonReply** [Azure/azure-dev#3907] [Q&A]
An [old issue](https://github.com/Azure/azure-dev/issues/1550) refers to `preinit`, but only in the context of removing it from the documentation, as does [this test](https://github.com/Azure/azure-dev/blob/6c00ad2d6c8eed7f364b344f1bc2d9ba5df0a485/cli/azd/pkg/ext/hooks_manager_test.go#L21) but otherwise it looks like neither are supported (anymore?)

I would like to be able to use preinit/postinit hooks to customise the one-time setup of a repo based on a sample that I publish, such as asking the user a number of setup questions that will then edit the files copied in by `azd init`. Is this kind of workflow supported or anticipated to be in the future?
* the postinit task might include removing itself, so that the resulting solution setup is unpolluted by concerns that relate purely to 'vending' the sample code and aren't relevant thereafter

**Comment by rajeshkamal5050** [comment]
@wbreza it this even supported?

**Reply by wbreza** [reply]
You are correct - the `preinit` hook is no longer supported.  Since the `azd init` command itself is used to download and initialize a template it would have inconsistent behavior depending on whether the `-t` template argument is specified, and whether the user already have the project downloaded locally.

Could you explain your scenario in a more detail so we can figure out next steps and see how it fits in with future scenarios on our roadmap?

**Reply by TWolversonReply** [reply]
I was thinking of a pattern where a template repository containing placeholders for resource names and the like would be post-processed once downloaded to the user's working copy, and parameters could then be substituted into the templated files. My earlier thinking also included providing both azdo and gha pipeline prototypes, and deleting the unused one automatically, since most users will use one or the other and not both. This particular case would be superseded by `azd pipeline config` which only includes the desired vendor format to start with.
